### PR TITLE
Let mongo use disk for really large queries.

### LINKED ISSUE #511: pulp2to3 migration fails to migrate docker_blob content due to aggregate mongo 100M limit
**Version**
0.11.6

Description of problem:
Having many docker_blobs, pulp2to3 migration fails with:

    Processing Pulp2 repositories, importers, distributors 3253/3283
    Pre-migrating Pulp 2 docker_blob content 10000/25670
    Initial Migration steps complete. Migration failed, You will want to investigate: https://satellite.example.com/foreman_tasks/tasks/fd24355e-9fea-4aab-8a49-4751e9dfb9ab rake aborted!
    ForemanTasks::TaskError: Task fd24355e-9fea-4aab-8a49-4751e9dfb9ab: Katello::Errors::Pulp3Error: Sort exceeded memory limit of 104857600 bytes

The reason is an aggregate method requires from mongo more memory than mongo limit is (100MB).

How reproducible:
100% in a scaled environment

Steps to Reproduce:
1. Sync 25k+ docker blobs to Pulp 2
2. Run pulp2to3 pre-migration

Actual results:
2 fails with the above error, plus /var/log/messages have backtrace:

Feb  8 10:44:06 satellite pulpcore-worker-1: pulp: rq.worker:ERROR: Traceback (most recent call last):
Feb  8 10:44:06 satellite pulpcore-worker-1: File "/usr/lib/python3.6/site-packages/rq/worker.py", line 936, in perform_job
Feb  8 10:44:06 satellite pulpcore-worker-1: rv = job.perform()
Feb  8 10:44:06 satellite pulpcore-worker-1: File "/usr/lib/python3.6/site-packages/rq/job.py", line 684, in perform
Feb  8 10:44:06 satellite pulpcore-worker-1: self._result = self._execute()
Feb  8 10:44:06 satellite pulpcore-worker-1: File "/usr/lib/python3.6/site-packages/rq/job.py", line 690, in _execute
Feb  8 10:44:06 satellite pulpcore-worker-1: return self.func(*self.args, **self.kwargs)
Feb  8 10:44:06 satellite pulpcore-worker-1: File "/usr/lib/python3.6/site-packages/pulp_2to3_migration/app/tasks/migrate.py", line 77, in migrate_from_pulp2
Feb  8 10:44:06 satellite pulpcore-worker-1: pre_migrate_all_content(plan)
Feb  8 10:44:06 satellite pulpcore-worker-1: File "/usr/lib/python3.6/site-packages/pulp_2to3_migration/app/pre_migration.py", line 70, in pre_migrate_all_content
Feb  8 10:44:06 satellite pulpcore-worker-1: pre_migrate_content_type(content_model, mutable_type, lazy_type, premigrate_hook)
Feb  8 10:44:06 satellite pulpcore-worker-1: File "/usr/lib/python3.6/site-packages/pulp_2to3_migration/app/pre_migration.py", line 124, in pre_migrate_content_type
Feb  8 10:44:06 satellite pulpcore-worker-1: pulp2_content_ids = premigrate_hook()
Feb  8 10:44:06 satellite pulpcore-worker-1: File "/usr/lib/python3.6/site-packages/pulp_2to3_migration/app/plugin/docker/utils.py", line 17, in find_tags
Feb  8 10:44:06 satellite pulpcore-worker-1: result = pulp2_models.Tag.objects.aggregate([sort_stage, group_stage1, group_stage2])
Feb  8 10:44:06 satellite pulpcore-worker-1: File "/usr/lib/python3.6/site-packages/mongoengine/queryset/base.py", line 1318, in aggregate
Feb  8 10:44:06 satellite pulpcore-worker-1: return collection.aggregate(final_pipeline, cursor={}, **kwargs)
Feb  8 10:44:06 satellite pulpcore-worker-1: File "/usr/lib64/python3.6/site-packages/pymongo/collection.py", line 2458, in aggregate
Feb  8 10:44:06 satellite pulpcore-worker-1: **kwargs)
Feb  8 10:44:06 satellite pulpcore-worker-1: File "/usr/lib64/python3.6/site-packages/pymongo/collection.py", line 2377, in _aggregate
Feb  8 10:44:06 satellite pulpcore-worker-1: retryable=not cmd._performs_write)
Feb  8 10:44:06 satellite pulpcore-worker-1: File "/usr/lib64/python3.6/site-packages/pymongo/mongo_client.py", line 1471, in _retryable_read
Feb  8 10:44:06 satellite pulpcore-worker-1: return func(session, server, sock_info, slave_ok)
Feb  8 10:44:06 satellite pulpcore-worker-1: File "/usr/lib64/python3.6/site-packages/pymongo/aggregation.py", line 148, in get_cursor
Feb  8 10:44:06 satellite pulpcore-worker-1: user_fields=self._user_fields)
Feb  8 10:44:06 satellite pulpcore-worker-1: File "/usr/lib64/python3.6/site-packages/pymongo/pool.py", line 694, in command
Feb  8 10:44:06 satellite pulpcore-worker-1: exhaust_allowed=exhaust_allowed)
Feb  8 10:44:06 satellite pulpcore-worker-1: File "/usr/lib64/python3.6/site-packages/pymongo/network.py", line 162, in command
Feb  8 10:44:06 satellite pulpcore-worker-1: File "/usr/lib64/python3.6/site-packages/pymongo/network.py", line 162, in command
Feb  8 10:44:06 satellite pulpcore-worker-1: parse_write_concern_error=parse_write_concern_error)
Feb  8 10:44:06 satellite pulpcore-worker-1: File "/usr/lib64/python3.6/site-packages/pymongo/helpers.py", line 168, in _check_command_response
Feb  8 10:44:06 satellite pulpcore-worker-1: max_wire_version)
Feb  8 10:44:06 satellite pulpcore-worker-1: pymongo.errors.OperationFailure: Sort exceeded memory limit of 104857600 bytes, but did not opt in to external sorting. Aborting operation. Pass allowDiskUse:true to opt in., full error: {'ok': 0.0, 'errmsg': 'Sort exceeded memory limit of 104857600 bytes, but did not opt in to external sorting. Aborting operation. Pass allowDiskUse:true to opt in.', 'code': 16819, 'codeName': 'Location16819'}

Expected results:
Clean pre-migration.

Additional info:
I guess /usr/lib/python3.6/site-packages/pulp_2to3_migration/app/plugin/docker/utils.py", line 17 should be:

result = pulp2_models.Tag.objects.aggregate([sort_stage, group_stage1, group_stage2], allowDiskUse=true)

### PR DESCRIPTION
**Pull request by ggainey**
fixes #511.
[nocoverage]

### CODE REVIEW

**Comment by ggainey** [comment]
@ipanova I have had zero luck with getting the 2to3 tessuite running locally, and was hoping to get a full test-run from CI. That appears, alas, to not be possible. If you can get the 2to3 tests to run against this PR, I would be most appreciative.

File: CHANGES/511.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Taught mongo-connection howto deal with Really Big Queries.
```
Comment (ipanova) [inline CHANGES/511.bugfix:1]: how to

**Review by ipanova — APPROVED** [review]
thanks!

**Comment by goosemania** [comment]
@ggainey , if you need any help or rubberduck to make tests run locally, let me know.

File: CHANGES/511.bugfix [hunk 2]
```diff
@@ -0,0 +1 @@
+Taught mongo-connection how to deal with Really Big Queries.
```
Comment (goosemania) [inline CHANGES/511.bugfix:1]: This should be a user-oriented message in the changelog, any error message which user will no longer see or anything which a user can relate to. This message sounds too implementation detail oriented to me.

**Comment by ipanova** [comment]
> @ggainey , if you need any help or rubberduck to make tests run locally, let me know.

we're working on fixing the ci.

**Pull request merged**